### PR TITLE
enable it test for delayed lines

### PR DESCRIPTION
**Pull request by dkhokhlov**
- enabled k8s_integration_test_for_dalyed_lines test

### CODE REVIEW

File: common/k8s/src/middleware/metadata.rs [hunk 1]
```diff
@@ -489,7 +490,9 @@ impl Middleware for K8sMetadata {
     ) -> Result<&'a dyn LineBufferMut, MiddlewareError> {
         // here we retry only pod log lines that do not have k8s metadata
         let file_name = line.get_file().unwrap_or("");
-        debug!("validate line from file: '{:?}'", file_name);
+        rate_limit!(rate = 1, interval = 5, {
```
Comment (jakedipity) [inline common/k8s/src/middleware/metadata.rs:493]: Do we need a rate limit for a debug log?

File: common/config/src/lib.rs [hunk 2]
```diff
@@ -190,6 +190,18 @@ impl Config {
...
+            // Check if the key starts with "LOGDNA_" or "MZ_"
+            if key.starts_with("LOGDNA_") || key.starts_with("MZ_") {
+                // Print the key and its value
+                if key.contains("KEY") || key.contains("PIN") {
+                    info!("{}: [REDACTED]", key);
+                } else {
+                    info!("{}: {}", key, value);
+                }
+            }
+        }
```
Comment (jakedipity) [inline common/config/src/lib.rs:203]: Printing all the env vars might leak some secrets besides the ones we recognize, and the agent already prints out its config values. Is this just a left over from testing?

File: bin/tests/it/k8s.rs [hunk 3]
```diff
@@ -1338,20 +1421,20 @@ async fn test_k8s_enrichment() {
         }
 
         // Wait for the data to be received by the mock ingester
-        tokio::time::sleep(tokio::time::Duration::from_millis(10_000)).await;
+        tokio::time::sleep(tokio::time::Duration::from_millis(3_000)).await;
```
Comment (jakedipity) [inline bin/tests/it/k8s.rs:1424]: Could this lower time cause integration tests to be unreliable?

File: bin/tests/it/k8s.rs [hunk 4]
```diff
@@ -1000,20 +1086,18 @@ fn get_agent_ds_yaml(
...
-                        existing_env_vars.push(serde_json::Value::Object(map));
-                    }
-                } else {
-                    panic!("unexpected");
-                }
+        let existing_env_vars = &mut cfgmap["spec"]["template"]["spec"]["containers"][0]["env"]
+            .as_array_mut()
+            .unwrap();
+        for env_var in env_vars {
+            if let Some((k, v)) = env_var.iter().next() {
```
Comment (jakedipity) [inline bin/tests/it/k8s.rs:1093]: nit: I find the previous method of converting `env_var` into a map and then collecting it to be a bit more ergonomic, but by no means is this bad/wrong/blocker.

File: bin/tests/it/k8s.rs [hunk 5]
```diff
@@ -200,6 +200,92 @@ async fn delete_service(client: Client, service_name: &str, namespace: &str) {
     }
 }
 
+async fn delete_daemonset(client: Client, daemonset_name: &str, namespace: &str) {
```
Comment (jakedipity) [inline bin/tests/it/k8s.rs:203]: I think we can do a better job to ensure RAII with Rust's language features. Maybe we instead have the `create_agent_ds` function return a `AgentGuard` which is comparable to a `MutexGuard` which we get after locking a mutex - this resource can implement these delete operations with the drop trait

**Review by jakedipity — COMMENTED** [review]
Mostly nit comments, but I had 1 big suggestion for ensuring RAII through a structured lifetime e.g. like a `MutexGuard`

File: common/config/src/lib.rs (hunk 2 above)
Comment (dkhokhlov) [reply common/config/src/lib.rs:203]: Requested from IBM support.

Comment (dkhokhlov) [reply common/config/src/lib.rs:203]: I am going do it in separate PR. that will not use open list.

File: bin/tests/it/k8s.rs (hunk 4 above)
Comment (dkhokhlov) [reply bin/tests/it/k8s.rs:1093]: test code. explicit easy to debug.

File: bin/tests/it/k8s.rs (hunk 3 above)
Comment (dkhokhlov) [reply bin/tests/it/k8s.rs:1424]: no,, it is enough. legacy attempts to fight with flakiness of test. now we properly wait for pod state.

File: bin/tests/it/k8s.rs (hunk 5 above)
Comment (dkhokhlov) [reply bin/tests/it/k8s.rs:203]: same as before - this is test code, explicit order etc, easy to debug. the Drop trait's drop method isn't asynchronous. so have to spawn new task. and then deal with borrow/ownership rules of cleanup params. complexity is increasing and new code will need to be tested itself.

**Review by jakedipity — APPROVED** [review]

**Pull request merged**